wpilibj/src/shared/java/edu/wpi/first/wpilibj/command/Scheduler.java

```diff
@@ -316,10 +278,10 @@ private void updateTable() {
       // Get the commands to cancel
       double[] toCancel = m_table.getNumberArray("Cancel", new double[0]);
       if (toCancel.length > 0) {
-        for (LinkedListElement e = m_firstCommand; e != null; e = e.getNext()) {
-          for (int i = 0; i < toCancel.length; i++) {
-            if (e.getData().hashCode() == toCancel[i]) {
-              e.getData().cancel();
+        for (Command command : m_commands) {
+          for (double cancelId : toCancel) {
+            if (command.hashCode() == cancelId) {
+              command.cancel();
```

Comment:
Commands should have a `final int getId()` method. It could delegate to `hashCode()` internally, or (better) `System.identityHashCode()`. Dunno how that would affect C++ though.